### ISSUE TITLE
no-loop-func not triggered for for/in loops

### ISSUE THREAD

**Issue by langdonx** [eslint/eslint#1573]
As titled. Code here:

```
(function () {
    'use strict';

    var i,
        funcHash = { f1: true, f2: true },
        funcs = {};

    for (i = 0; i < 10; i += 1) {
        funcs[i] = function () {
            return i;
        };
    }

    for (i in funcHash) {
        if (funcHash.hasOwnPropery(i)) {
            funcs[i] = function () {
                return i;
            };
        }
    }
}());

```

And the output:

```
$ eslint --version
v0.10.1

$ eslint test.js

test.js
  9:14  warning  Don't make functions within a loop  no-loop-func

✖ 1 problem
```

I would expect the for/in loop to generate a warning as well.  Both JSLint and JSHint warn in this circumstance.
